managed/CounterStrikeSharp.API/Core/Memory/MemoryManager.cs

```diff
@@ -0,0 +1,318 @@
+                                               "  stats - Print garbage collector statistics.\n" +
+                                               "  start - Starts the memory manager that handles leaking resources.\n" +
+                                               "  stop - Stops the memory manager.\n" +
+                                               "  pause - Stops the memory manager.\n" +
+                                               "  resume - Resumes the memory manager.\n");
+                } break;
+            }
+        }
+
+        public void Load()
+        {
+            _commandManager.RegisterCommand(new("css_memory", "Counter-Strike Sharp Memory Manager options.",
+                OnCommand)
+            {
+                ExecutableBy = CommandUsage.CLIENT_AND_SERVER,
+                MinArgs = 1,
+                UsageHint = "[option]\n" +
+                            "  stats - Print garbage collector statistics.\n" +
+                            "  start - Starts the memory manager that handles leaking resources.\n" +
+                            "  stop - Stops the memory manager.\n" +
+                            "  pause - Stops the memory manager.\n" +
+                            "  resume - Resumes the memory manager.\n",
+            });
+
+            Start();
+        }
+
+        private void PrintStatistics(CommandInfo info)
+        {
+            info.ReplyToCommand("Memory Manager Statistics:\n" +
+                                      $"State: {State}\n" +
+                                      $"Total Released: {TotalReleased}\n" +
+                                      $"Last Released: {LastReleased}\n" +
+                                      $"Current Resources: {CurrentResources}\n" +
+                                      $"Last Updated: {LastUpdated.ToString("yyyy.MM.dd hh:mm:ss tt")} (UTC)\n" +
+                                      $"Heap Memory Usage: ~{TotalMemoryMB:F5} MB ({TotalMemory} bytes)\n" +
+                                      $"Total Allocated Bytes: ~{TotalAllocatedMB:F5} MB ({TotalAllocated} bytes)");
+        }
+
+        public void Start()
+        {
+            if (CoreConfig.EnableMemoryManager)
+            {
+                if (State == MemoryManagerState.Idle || State == MemoryManagerState.InProgress)
+                {
+                    _logger.LogInformation("Service is already running");
+                    return;
+                }
+
+                if (State == MemoryManagerState.Halted)
+                {
+                    _logger.LogInformation("Service should be resumed, not started");
+                    return;
+                }
+
+                _logger.LogInformation("Starting service...");
+
+                try
+                {
+                    _thread.Start();
+                } catch (ThreadStateException)
+                {
+                    _thread = new Thread(BackgroundThread);
+                    _thread.Start();
```

Comment:
this handles thread creation, if it was terminated earlier this exception is thrown